test/src/org/exist/storage/ModificationTimeTest.java

```diff
@@ -0,0 +1,209 @@
+            
+            binaryDoc = storeBinary(filename, data, mimeType);
+            assertNotNull(binaryDoc);
+
+            long modificationTimeAfter = binaryDoc.getMetadata().getLastModified();
+            //check the mimetype has been preserved across database restarts
+            assertNotEquals(modificationTimeBefore, modificationTimeAfter);
+    }
+    
+     /**
+     * Store a valid XML resource, wait for a while and then overwrite it 
+     * with another valid XML resource. The resource's modification time should
+     * have been updated afterwards.
+     */
+    @Test
+    public void check_if_modification_time_is_updated_xml() throws EXistException, InterruptedException, PermissionDeniedException, LockException, IOException, TriggerException, SAXException, DatabaseConfigurationException {
+            
+            IndexInfo info = storeXML(XML_FILENAME, VALID_XML);
+            assertNotNull(info);
+            DocumentImpl doc = info.getDocument();
+            
+            long modificationTimeBefore = doc.getMetadata().getLastModified();
+            Thread.sleep(500);
+            
+            info = storeXML(XML_FILENAME, VALID_XML);
+            assertNotNull(info);
+            doc = info.getDocument();
+            
+            long modificationTimeAfter = doc.getMetadata().getLastModified();
+            
+            assertNotEquals(modificationTimeBefore, modificationTimeAfter);
+    }
+    
+     /**
+     * Store a valid XML resource, wait for a while and then try to overwrite
+     * it with an invalid XML resource. The invalid XML should be rejected and 
+     * the resource's modification time should be the same afterwards.
+     */
+    @Test
+    public void check_if_modification_time_is_not_updated_on_parse_error() throws EXistException, InterruptedException, PermissionDeniedException, LockException, IOException, TriggerException, SAXException, DatabaseConfigurationException {
+        
+            
+            IndexInfo info = storeXML(XML_FILENAME, VALID_XML);
+            assertNotNull(info);
+            DocumentImpl doc = info.getDocument();
+            final XmldbURI docUri = doc.getFileURI();
+            
+            long modificationTimeBefore = doc.getMetadata().getLastModified();
+            
+            Thread.sleep(500);
+            
+            boolean threw = false;
+            info = null;
+            try {
+                info = storeXML(XML_FILENAME, INVALID_XML);
+            } catch (SAXException e) {
+                threw = true;
+            }
+            assertTrue(threw);
+            assertNull(info);
+            
+            doc = getDocument(docUri);
+            assertNotNull(doc);
+            
+            long modificationTimeAfter = doc.getMetadata().getLastModified();
+            assertEquals(modificationTimeBefore, modificationTimeAfter);
+    }
+
+   protected BrokerPool startDB() throws DatabaseConfigurationException, EXistException {
```

Comment:
Could this not have an `@Before` annotation and be called once per-test, or better yet if it's possible then `@BeforeClass` so that the database only need be started once for the suite of tests (you would also need to change your `@After` to `@AfterClass`.